evdev/src/Evdev/Codes.chs

```diff
@@ -9,7 +11,37 @@ seems to be on its way to being fixed with `.hie` files (enable `-fwrite-ide-inf
 
 -- | Datatypes corresponding to the constants in [input-event-codes.h](https://github.com/torvalds/linux/blob/master/include/uapi/linux/input-event-codes.h).
 -- See [the Linux Kernel documentation](https://www.kernel.org/doc/html/latest/input/event-codes.html) for full details, noting that all names have been mechanically transformed into CamelCase.
-module Evdev.Codes where
+module Evdev.Codes 
+    ( EventType(..)
+    , SyncEvent(..)
+    , Key
+        ( ..
+        , KeyHanguel
+        , KeyCoffee
+        , KeyDirection
+        , KeyBrightnessZero
+        , KeyWimax
+        , BtnMisc
+        , BtnMouse
+        , BtnTrigger
+        , BtnGamepad
+        , BtnSouth
+        , BtnEast
+        , BtnNorth
+        , BtnWest
+        , BtnDigi
+        , BtnWheel
+        , KeyBrightnessToggle
+        , BtnTriggerHappy )
+    , RelativeAxis(..)
+    , AbsoluteAxis(..)
+    , SwitchEvent(..)
+    , MiscEvent(..)
+    , LEDEvent(..)
+    , RepeatEvent(..)
+    , SoundEvent(..)
+    , DeviceProperty(..)
+    ) where
```

Comment:
Is there a reason for switching to an explicit list in this commit? It looks like we're still exporting everything.

Reply:
Apart from being better practice, it allows you to import the enum _and_ the synonyms by importing just `Key(..)`.

Reply:
> being better practice

Yeah, I'm aware of that, but sometimes it seems like unnecessary bureaucracy when you just want to export everything.

> it allows you to import the enum _and_ the synonyms by importing just `Key(..)`

Aha, I see. I've never really used `PatternSynonyms` much.